repository/firestore/firestore.go

```diff
@@ -69,9 +71,65 @@ func (r *firestoreRepository[T]) Update(data interface{}, filters ...repository.
 	return errors.ErrMethodNotImplemented
 }
 
-// Delete is not implemented.
-func (r *firestoreRepository[T]) Delete(filters ...repository.Filter) error {
-	return errors.ErrMethodNotImplemented
+// Delete deletes all the entities that match the given options. This method is not responsible for performing soft
+// deletes.
+//
+// Any project using this repository must implement soft deletion at the firestore-level if they're in need of soft
+// deletes. Consider using something like https://extensions.dev/extensions/adamnathanlewis/ext-firestore-soft-deletes
+// We DO NOT recommend any third-party extension, and they're only presented here as an example of what can be used
+// to implement soft deletes.
+//
+// Delete does not remove all the records at once, it will perform the document removal in small batches. This mechanism
+// prevents having out-of-memory errors.
+func (r *firestoreRepository[T]) Delete(options ...repository.Option) error {
+	ctx := context.Background()
+	col := r.client.Collection(r.Model().TableName())
+	r.applyOptions(&col.Query, options...)
+
+	err := r.deleteBatch(ctx, col, 30)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// deleteBatch is a helper function that allows deleting documents in small batches of the given size.
+func (r *firestoreRepository[T]) deleteBatch(ctx context.Context, col *firestore.CollectionRef, size int) error {
+	writer := r.client.BulkWriter(ctx)
+	for {
+		// Get a batch of documents
+		iter := col.Limit(size).Documents(ctx)
+
+		// Initialize the delete counter to 0
```

Comment:
This comment is evident from the expression it is documenting. Try to explain the motivation behind setting this counter to 0.

```suggestion
		// Track the number of deleted records in this batch
```